### PR TITLE
feat(oot framework): allow model ops sample tag markers

### PR DESCRIPTION
**Pull request by anubhavjana**
Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [ ] bug
- [x] feature
- [ ] documentation
- [ ] cleanup

## Problem
- `edits.ops.include[].tags` in model-ops YAML configs (e.g. `torch.nn.functional.embedding.1_spyre`) were runtime-only labels, only used for the `[TAGS = ...]` report line and JUnit XML sidecar.

- These tags were never attached as pytest marks, so no `-m` expression could select an individual sample --- only the coarser `op__<name>` marker (whole op, index stripped) was selectable.

## Feature
- `_OOTOpMarkerPatcher` now applies each `ModelOpInfo.op_tags` entry as its own pytest mark alongside the existing `op__`/`dtype__` marks, enabling `-m "torch.nn.functional.embedding.1_spyre"` to select a single sample.

### Validation:

```bash
bash tests/run_test.sh tests/configs/model_ops_tests/granite-3.3-8b-instruct_spyre.yaml -v -m "torch.nn.functional.embedding.1_spyre" --collect-only -q
```

now correctly collect test by ops.include.tags

image

### CODE REVIEW

**Review by ashokponkumar — APPROVED** [review]

**Review by kiszk — APPROVED** [review]
Thank you. Confirmed at my end, too.
